src/prefect/server/schemas/actions.py

```diff
@@ -41,6 +41,12 @@ def validate_block_document_name(value):
     return value
 
 
+def validate_artifact_key(value):
+    if not bool(re.match(LOWERCASE_LETTERS_AND_DASHES_ONLY_REGEX, value)):
+        raise ValueError("Artifact name must only contain lowercase letters, numbers, and dashes")
```

Comment:
That's now how the interface is here. Users don't know (or have to know) it's called key. But if this is what we want, should I change this to reflect it?

https://github.com/PrefectHQ/prefect/pull/8832

Reply:
Why are we calling it a key internally if it's a name? :D

Reply:
Ah shoot, I should have caught that. I think we should standardize on `key` everywhere, including the SDK.

Reply:
I don't know the reasoning around why key was chosen. It's an optional field, so.. I suppose key implies that it can be used to retrieve something. Name sounded more intuitive to me versus having 2 new items (what's an artifact? what's a key? is this an mtg reference?) for users to understand. But I ultimately defer to product.